Exercise/src/my_string.cpp

```diff
@@ -0,0 +1,80 @@
+#include "my_string.h"
+#include <cstddef>
+
+std::size_t Exercise_1::CMyString::GetStringLength(const char *cpszStringToCount) noexcept {
+	std::size_t nStringLength = 0;
+	char cCurrentChar = cpszStringToCount[0];
+	while(cCurrentChar != '\0')
+	{
+		++nStringLength;
+		cCurrentChar = cpszStringToCount[nStringLength];
+	}
+	// Adding 1 because string is zero terminated
+	return nStringLength + 1;
+}
+
+void Exercise_1::CMyString::CopyString(const char *cpszStringToCopy, char *pszStringToPaste, std::size_t nCountOfChars)
+{
+	for(std::size_t nIndex = 0; nIndex < nCountOfChars; nIndex++)
+	{
+		pszStringToPaste[nIndex] = cpszStringToCopy[nIndex];
+	}
+}
+
+Exercise_1::CMyString::CMyString(const char* cszCharsSequence) 
+{
+	m_nSize = GetStringLength(cszCharsSequence);
+	m_szData = new char[m_nSize];
```

Comment:
Can new throw an error?

Reply:
How should I resolve such things in constructors or it should be moved somewhere else?

Reply:
Created TryToAllocate function and fixed